h/sql_tasks/sql_query.py

```diff
@@ -0,0 +1,60 @@
+import textwrap
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from typing import Optional
+
+import sqlalchemy as sa
+from sqlalchemy.orm import Session
+from tabulate import tabulate
+
+
+@dataclass
+class SQLQuery:
+    """A class representing an individual SQL query."""
+
+    index: int
+    """Index of this query inside the script."""
+
+    text: str
+    """Text of the query."""
+
+    start_time: datetime = None
+    """Time execution began."""
+
+    duration: timedelta = None
+    """Duration of query execution."""
+
+    columns: Optional[list] = None
+    """Columns of the returned values (if any)."""
+
+    rows: Optional[list] = None
+    """Rows of the returned values (if any)."""
+
+    def execute(self, db_session: Session):
+        """Execute this query in the given session."""
+
+        self.start_time = datetime.now()
+
+        cursor = db_session.bind.execute(sa.text(self.text))
+        if cursor.returns_rows:
+            self.columns = cursor.cursor.description
+            self.rows = cursor.fetchall()
+
+        self.duration = datetime.now() - self.start_time
```

Comment:
Nice we could send this to newrelic in a `trace()` method or similar and have a graph of how long it takes over time.

Something for an enhancement pass after we have the first version working though.